rpm/files_test.go

```diff
@@ -0,0 +1,76 @@
+package rpm
+
+import (
+	"context"
+	"testing"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/test"
+	"github.com/quay/zlog"
+)
+
+var rpmFilesTestcases = []struct {
+	name     string
+	isRPM    bool
+	filePath string
+}{
+	{
+		name:     "Found Node",
+		isRPM:    true,
+		filePath: "usr/lib/node_modules/npm/node_modules/safe-buffer/package.json",
+	},
+	{
+		name:     "Found Python",
+		isRPM:    true,
+		filePath: "usr/lib64/python3.9/site-packages/libcomps-0.1.18-py3.9.egg-info/PKG-INFO",
+	},
+	{
+		name:     "Not found",
+		isRPM:    false,
+		filePath: "usr/lib/node_modules/npm/node_modules/safe-buffer/package.jsonx",
+	},
+}
+
+func TestIsRPMFile(t *testing.T) {
+	ctx := zlog.Test(context.Background(), t)
+	a := test.NewCachedArena(t)
+	layer := test.LayerRef{
+		Registry: "registry.access.redhat.com",
+		Name:     "ubi9/nodejs-18",
+		Digest:   `sha256:1ae06b64755052cef4c32979aded82a18f664c66fa7b50a6d2924afac2849c6e`,
+	}
+	a.LoadLayerFromRegistry(ctx, t, layer)
+	r := a.Realizer(ctx).(*test.CachedRealizer)
+	t.Cleanup(func() {
+		if err := r.Close(); err != nil {
+			t.Error(err)
+		}
+	})
+
+	realizedLayers, err := r.RealizeDescriptions(ctx, []claircore.LayerDescription{
+		{
+			Digest:    layer.Digest,
+			URI:       "http://example.com",
+			MediaType: test.MediaType,
+			Headers:   make(map[string][]string),
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, tt := range rpmFilesTestcases {
+		ctx, cancel := context.WithCancel(zlog.Test(context.Background(), t))
```

Comment:
probably doesn't matter much that this is outside the `Run`, right? As in, is this here for a particular reason or just because it doesn't matter?

Reply:
It makes it a little more interesting that the context is cancelled in a different goroutine but I don't think it really adds to the validity of the test, I can move it inside for readability